README.md

```diff
@@ -139,12 +139,28 @@ See [here](https://javadoc.io/doc/dev.openfeature/sdk/latest/) for the Javadocs.
 Look [here](https://openfeature.dev/ecosystem?instant_search%5BrefinementList%5D%5Btype%5D%5B0%5D=Provider&instant_search%5BrefinementList%5D%5Btechnology%5D%5B0%5D=Java) for a complete list of available providers.
 If the provider you're looking for hasn't been created yet, see the [develop a provider](#develop-a-provider) section to learn how to build it yourself.
 
-Once you've added a provider as a dependency, it can be registered with OpenFeature like this:
+Once you've added a provider as a dependency, you can register it with OpenFeature in either an asynchronous or synchronous manner. Below are examples of how to set providers using both methods:   
+   
+#### Asynchronous Provider Registration  
+To register a provider asynchronously, ensuring it is initialized before further actions are taken, you can use the `setProviderAndWait` method as shown in the following example:  
 
 ```java
-    OpenFeatureAPI.getInstance().setProvider(new MyProvider());
+    OpenFeatureAPI api = OpenFeatureAPI.getInstance();
+    api.setProviderAndWait(new MyProvider());
+```
+
+This method is appropriate when you need to wait for the provider's initialization to complete before proceeding.  
+
+#### Synchronous Provider Registration  
+For synchronous provider registration, where you set the provider without waiting for initialization, you can use the `setProvider` method directly:  
+
+```java
+    OpenFeatureAPI api = OpenFeatureAPI.getInstance();
+    api.setProvider(new MyProvider());
 ```
 
+This is a straightforward synchronous operation, suitable when you don't need to wait for the provider to initialize before continuing.   
+
```

Comment:
I don't think this terminology is quite correct (or at least it's confusing). I would instead use the word "blocking". `setProviderAndWait` _blocks_ until the provider is ready, while `setProvider` does not.